src/main/java/org/sagebionetworks/bridge/spring/controllers/AuthenticationController.java

```diff
@@ -85,14 +85,18 @@ public JsonNode emailSignIn() {
         verifySupportedVersionOrThrowException(app);
         
         CriteriaContext context = getCriteriaContext(app.getIdentifier());
+
         UserSession session = null;
         try {
             session = authenticationService.emailSignIn(context, signInRequest);
-        } catch(ConsentRequiredException e) {
-            setCookieAndRecordMetrics(e.getUserSession());
+        } catch (ConsentRequiredException e) {
+            session = e.getUserSession();
+            updateRequestInfoFromSession(session);
             throw e;
+        } finally {
+            request().setAttribute("CreatedUserSession", session);
         }
-        setCookieAndRecordMetrics(session);
+        updateRequestInfoFromSession(session);
```

Comment:
The problem is, `authService.emailSignIn` doesn't only throw `ConsentRequiredException`, it also throws `AuthenticationFailedException` and `EntityNotFoundException`. In the original code, the try/catch block only handles `CRE` instead of other exceptions, so it calls `setCookieAndRecordMetrics` twice rather than in finally block. (For other two Exceptions, no session info is recorded)

Reply:
With regard to other comments below, I'll integrate `updateRequestInfoFromSession` along with setting request attributes together in `AuthController`, so that we don't need to grab `e.getUserSession()` again in Filters.